rewrite-properties/src/main/java/org/openrewrite/properties/AddProperty.java

```diff
@@ -59,11 +59,11 @@ public String getDescription() {
     }
 
     @Override
-    public TreeVisitor<?, ExecutionContext> getVisitor() {
+    public PropertiesVisitor<ExecutionContext> getVisitor() {
```

Comment:
Why make this more specific?

Reply:
In #3276 it was requested if it's possible to reuse `AddProperty`. That's awkward right now as `getVisitor()` returns a generic `TreeVisitor`, and the `visitFile()` within that returns a `Properties` instead of `Properties.File`.

_(taken from above, in case you'd missed it; if my description is not clear do please tell!)_

Reply:
(2) is the purpose of `PropertiesIsoVisitor` vs. `PropertiesVisitor`.

Reply:
Ah didn't realize we had an iso visitor for properties as well; that's indeed better. I'll clean this up when I find a moment.

Reply:
Now we only switch to `PropertiesIsoVisitor` and change the `getVisitor()` return type. Much better.